[Cheng Zihao] ip

chengzihao123 opened the pull request:
### ChattyBuddy
> “I remind myself that I’m always more satisfied by human interaction than by a digital connection.” - Maulik Pancholy

ChattyBuddy is a excellent chat buddy to any users. It is ~~beyond just a digital bot~~ but a human friend that dedicates to remember things the user needs to do including:

- [x] 1. Todo task 
- [x] 2. Deadline task (task with a due date)
- [x] 3. Event task (task with a start and an end date)

And there is a vast pools of commands that the user could use including:

- list
- todo xxx
- deadline xxx /by xxx
- event xxx /from xxx /to xxx
- mark 
- unmark
- delete
- find xxx

And it is completely **FREE**!!

All you need to do is:

1. Download it from [here](https://github.com/chengzihao123/ip)
2. Scroll down to find the download manual
3. Click it and select a directory
4. Open the application on your computer and have a chat with it! 😎  

---
Here is my Main class: 
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chengzihao123 committed:
added name to bot

chengzihao123 committed:
change name to chattyBuddy

chengzihao123 committed:
added echo feature

chengzihao123 committed:
added list feature

chengzihao123 committed:
added mark as done feature

chengzihao123 committed:
added different events

chengzihao123 committed:
added new classes

chengzihao123 committed:
added classes

chengzihao123 committed:
added basic error handling

chengzihao123 committed:
added more exceptions

chengzihao123 committed:
added delete function

chengzihao123 committed:
added test io inputs

chengzihao123 committed:
added save feature

chengzihao123 committed:
refactored different task classes

chengzihao123 committed:
Merge branch 'branch-level-7'

chengzihao123 committed:
Add date recognisaton for deadline class

chengzihao123 committed:
Add date time conversion for Event class

chengzihao123 committed:
Merge branch 'branch-level-8'

chengzihao123 committed:
Add more oop classes

chengzihao123 committed:
Merge tag 'A-MoreOOP'

chengzihao123 committed:
Add different packages

chengzihao123 committed:
Merge branch 'A-Packages'

chengzihao123 committed:
Merge branch 'add-gradle-support' of https://github.com/chengzihao123/ip into add-gradle-support

chengzihao123 committed:
Merge branch 'add-gradle-support'

chengzihao123 committed:
Add gradle to project

chengzihao123 committed:
Add test cases for parser, deadline and event class

chengzihao123 committed:
Merge branch 'A-Junit'

chengzihao123 committed:
Add JavaDoc to classes and methods

chengzihao123 committed:
Refactor code to comply with coding standard

chengzihao123 committed:
Add find feature

chengzihao123 committed:
Merge branch 'branch-A-JavaDoc'

chengzihao123 committed:
Fix merge conflicts between master and branch-A-CodingStandard

chengzihao123 committed:
Fix merge conflicts between master and branch-Level-9 branch

chengzihao123 committed:
Fix spacing and constant names

Jalenleeruixian commented on src/main/java/main/ChattyBuddy.java:
```diff
@@ -0,0 +1,61 @@
+ * such as UI, storage, and task lists management.
+ */
+public class ChattyBuddy {
+
```
Good use of java docs

Jalenleeruixian commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,264 @@
+     * @throws EmptyTaskException    if the user attempts to add a task without providing a description.
+     * @throws TaskIndexOutOfBound   if the user provides an index for a task that does not exist.
+     */
+    public static void parseCommand(String command, TaskList taskList, Ui ui, Storage storage) throws InvalidInputException, EmptyTaskException, TaskIndexOutOfBound {
```
Naming convention of methods adhered.

Jalenleeruixian commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,61 @@
+ * The Storage class controls the storage management
+ * It contains methods to load tasks from a file and also to save user inputs to the file
+ */
+public class Storage {
```
Good use of OOP.

Jalenleeruixian commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,92 @@
+    protected boolean isDone;
+    protected String type;
+
+    /**
```
Java docs usage is good

Jalenleeruixian commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,93 @@
+        TASKS.get(index).setMarkStatus(true);
+    }
+
+    /**
```
Good use of Java docs

Jalenleeruixian reviewed (commented):
Overall, the coding standard was adhered to.

chengzihao123 committed:
Fix styling issues in the Main directory

chengzihao123 committed:
Merge branch 'A-CheckStyle'

wubojin commented on src/main/java/exceptions/TaskIndexOutOfBound.java:
```diff
@@ -0,0 +1,18 @@
+ * The TaskIndexOutOfBound class is a custom exception that is thrown when the user
+ * attempts to refer to a task with an invalid index (i.e., the task does not exist).
+ */
+public class TaskIndexOutOfBound extends Exception {
```
Exception classes are descriptive but consider Java's naming conventions for exceptions eg. `TaskIndexOutOfBoundException` instead of `TaskIndexOutOfBound`

wubojin commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,284 @@
+    public static void parseCommand(
+            String command, TaskList taskList, Ui ui, Storage storage
+    ) throws InvalidInputException, EmptyTaskException, TaskIndexOutOfBound {
+        String[] slicedStr = command.split(" ");
```
Perhaps could use a better variable name to improve clarity, eg. `stringParts` instead of `slicedStr`

wubojin commented on src/test/java/ui/UiStub.java:
```diff
@@ -0,0 +1,53 @@
+
+import task.Task;
+import java.util.ArrayList;
+public class UiStub extends Ui {
```
Good use of stub!

wubojin commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,155 @@
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
```
Still hello from Duke?

wubojin reviewed (commented):
Overall code quality is good with clear descriptive names, and consistent formatting helps with readability.

chengzihao123 committed:
Add the GUI for ChattyBuddy without changes to the test files

chengzihao123 committed:
Change the logo from duke to chattybuddy

chengzihao123 committed:
Refactor the code for test files to correspond to the addition of GUI instead of text UI

chengzihao123 committed:
Change the icon for both user and the bot

chengzihao123 committed:
Merge branch 'branch-Level-10'

apollo-tan commented on src/main/java/exceptions/EmptyTaskException.java:
```diff
@@ -0,0 +1,18 @@
+        super(String.format("OOPS!! The description of a %s cannot be empty %n"
+                             + "Have you forgotten to enter the details?", task));
+    }
+}
```
Very detailed class and method headers!

apollo-tan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,298 @@
+    public static String parseCommand(
+            String command, TaskList taskList, Storage storage
+    ) throws InvalidInputException, EmptyTaskException, TaskIndexOutOfBound {
+        String[] slicedStr = command.split(" ");
```
I feel that the `slicedStr` could be renamed to something like `commandParts` or `inputParts` to emphasise that it's the user command + arguments being broken down into different parts for processing purposes.

apollo-tan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,298 @@
+    public static Task parseSavedData(String[] dataArr) {
+        Task task;
+        switch (dataArr[0]) {
+        case "T":
```
Consider indenting your case statements for improved readability

apollo-tan commented on src/main/java/ui/MainWindow.java:
```diff
@@ -0,0 +1,99 @@
+    public void showGoodbyeMessage() {
+        addDialog(DialogBox.getBotDialog("Bye. Hope to see you again soon!", botImage));
+    }
+}
```
Overall well done! Code mostly adheres to the java coding standard.

chengzihao123 committed:
Add assertions for some classes

chengzihao123 committed:
Merge pull request #1 from chengzihao123/branch-A-Assertions

Add assertions for some classes

chengzihao123 committed:
Refactor the code to improve code quality

chengzihao123 committed:
Modify code quality of current code files

There are several places where the code quality

has been improved such as the Deadline and Event classes

Such improvements include extracting out constants

to avoid magic literal

This allows the code to be more readable and to avoid errors

chengzihao123 committed:
merge master with A-CodeQuality

chengzihao123 committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/chengzihao123/ip into branch-A-CodeQuality

chengzihao123 committed:
Refactor Parser class to utilize Java Streams for task operations

Previously, the task operations in the Parser class were using
traditional loops and conditionals for operations like finding matching
tasks. This approach, whiel fucntional, cna be made mroe concise by
leveraging streams.

Changes made: replaced traiditonal for-loops with streams in method like
'handleFindCommand' to filter and collect matching tasks based on
keywords.

This refactorign enhances the code by reducing verbosity, improving
readability, and following modern java best practices.                                                                         ~

chengzihao123 committed:
Refactor Parser class to utilize Java Streams for task operations

Previously, the task operations in the Parser class were using
traditional loops and conditionals for operations like finding matching
tasks. This approach, whiel fucntional, cna be made mroe concise by
leveraging streams.

Changes made: replaced traiditonal for-loops with streams in method like
'handleFindCommand' to filter and collect matching tasks based on
keywords.

This refactorign enhances the code by reducing verbosity, improving
readability, and following modern java best practices.
~                                                                         ~

chengzihao123 committed:
Add streams to parser methods

chengzihao123 committed:
Minor change to the configuration of the code

chengzihao123 committed:
Refactor Parser class to utilize Java Streams for task operations

Previously, the task operations in the Parser class were using loops and conditionals for operations like finding matching tasks and processing lists of tasks. This approach, while functional, can be made more concise by using streams.

Changes made: Replaced traditional for-loops with Java Streams in methods such as
  handleFindCommand to filter and collect matching tasks based on keywords.

This refactoring enhances the code by reducing verbosity, improving
readability, and following modern Java best practices.

chengzihao123 committed:
Fix merge conflict

chengzihao123 committed:
Merge branch 'branch-A-Streams' of https://github.com/chengzihao123/ip into branch-A-Streams

chengzihao123 committed:
Refactor Parser class to utilize Java Streams for task operations

Previously, the task operations in the Parser class were using traditional
loops and conditionals for operations like finding matching tasks and
processing lists of tasks. This approach, while functional, can be made
more concise and readable by leveraging Java Streams.

Changes made:
- Replaced traditional for-loops with Java Streams in methods such as
  `handleFindCommand` to filter and collect matching tasks based on keywords.

This refactoring enhances the code by reducing verbosity, improving
readability, and following modern Java best practices.

chengzihao123 committed:
Add undo method skeleton

chengzihao123 committed:
Merge branch 'branch-A-CodeQuality'

chengzihao123 committed:
Implement the undo function

chengzihao123 committed:
Merge branch 'branch-C-Undo'

chengzihao123 committed:
Set up continuous integration

chengzihao123 committed:
Rename the main java file name to Duke

chengzihao123 committed:
Add javadoc to test files and stubs

chengzihao123 committed:
refactor code

chengzihao123 committed:
Delete redundant test case

chengzihao123 committed:
Add user guide

chengzihao123 committed:
Modify and complete the user guide
